hoomd/md/pair/aniso.py

```diff
@@ -563,3 +572,425 @@ def type_shapes(self):
+
+    Patchy potentials compute the  potential between particles with
+    angular patches with an implementation based on
+    `Beltran-Villegas et. al.`_. With distance between particles
+    :math:`r` and minimal angles between respective particle direction and
+    the displacement vector :math:`\theta_i`, :math:`\theta_j` the patchy potential
+    computes:
+
+    .. _Beltran-Villegas et. al.: https://dx.doi.org/10.1039/C3SM53136H
+
+    .. math::
+
+        U(r, \theta_i, \theta_j) = f(\theta_i, \alpha, \omega)
+        f(\theta_j, \alpha, \omega)
+        U_{\mathrm{pair}}(r)
+
+    where :math:`f` is an orientation-dependent factor of the patchy spherical
+    cap half-angle :math:`\alpha` and patch steepness :math:`\omega`.
+
+    .. math::
+        \begin{align}
+        f(\theta, \alpha, \omega) &= \frac{\big(1+e^{-\omega (\cos{\theta} -
+        \cos{\alpha}) }\big)^{-1} - f_{min}}{f_{max} - f_{min}}\\
+        f_{max} &= \big( 1 + e^{-\omega (1 - \cos{\alpha}) } \big)^{-1} \\
+        f_{min} &= \big( 1 + e^{-\omega (-1 - \cos{\alpha}) } \big)^{-1} \\
+        \end{align}
+
+    .. invisible-code-block: python
+        patchy = hoomd.md.pair.aniso.PatchyLJ(nlist = neighbor_list,
+                                              default_r_cut = 3.0)
+        pair_params = {'epsilon': 10, 'sigma': 1}
+
+        patchy.params.default = dict(pair_params = pair_params,
+                                     envelope_params = {'alpha': math.pi/4,
+                                                        'omega': 30})
+        patchy.patches.default = []
+        simulation.operations.integrator.forces = [patchy]
+
+    For multiple patches, every combination of patch and
+    We use PatchyLJ as the example.
+
+    To make specific patches
+
+    todo write example
+
+    .. py:attribute:: params
+
+        The Patchy potential parameters. A dictionary of dictionaries that
+        are passed on to the envelope and to the pair potential.
+
+        The dictionary has the following keys:
+
+        * ``envelope_params`` (`dict`, **required**)
+
+          * ``alpha`` (`float`) - patch half-angle :math:`\alpha` :math:`[\mathrm{rad}]`
+          * ``omega`` (`float`) - patch steepness :math:`\omega`
+
+
+        * ``pair_params`` (`dict`, **required**) -
+          passed to isotropic potential (see subclasses).
+
+        .. rubric:: Example:
+
+        .. code-block:: python
+
+            envelope_params = {'alpha': math.pi/4, 'omega': 30}
+            patchy.params[('A', 'A')] = dict(pair_params = pair_params,
+                                             envelope_params = envelope_params)
+
+        Type: `TypeParameter` [`tuple` [``particle_type``, ``particle_type``],
+        `dict`]
+
+    .. py:attribute:: patches
+
+        List of vectors pointing to patch centers. Unnormalized vectors are normalized.
+        If a particle type does not have patches, set the patches to an empty list.
```

Comment:
If a particle does not have patches is the whole particle sticky?

Reply:
No, then `Pathcy*` provides no interaction at all. You also need to apply `pair.LJ` to make the whole particle volume excluding (or sticky if needed).

We should clarify this by writing the patch interaction as a double sum over the patches. When either sum is over 0 patches, then the whole sum is 0.

Reply:
Should we even allow users to make another particle patchless? What would be the purpose of that since they are not interacting at all in that case?

Reply:
The assumption is that you use a volume-excluding potential in addition to a `Pathcy` one. Otherwise, even your patch particles can pass through each other when the patches are _not_ aligned.

One example where users may want no patches: Explicit solvent simulations with an isotropic solvent and patchy particles:
```
wca = hoomd.md.pair.LJ(nlist=..., default_r_cut=2**(1.0/6.0))
wca.params.default= {'sigma': 1.0, 'epsilon': 1.0}

patchy_lj = hoomd.md.pair.ansio.PatchyLJ(nlist=..., default_r_cut=2.5)
patchylj.params.default = dict(pair_params = {'sigma': 1.0, 'epsilon': 1.0},
                               envelope_params = envelope_params)

patchy_lj.patches['A'] = [(1,0,0), (-1,0,0)]
patchy_lj.patches['solvent'] = []
```

Here `wca` provides the hard core repulsion. `patchy_lj` adds attractive patches on top of that for _A-A_ particle interactions. The `patchy_lj` energy is 0 for _A-solvent_ and _solvent-solvent_ particle pairs.

Reply:
This should probably be documented?

Reply:
I don't think it belongs in the reference documentation., but it could make a nice how-to guide.

Reply:
This would be good material for an eventual tutorial.